library/datasort.c

```diff
@@ -903,6 +910,12 @@ static struct datasort_chunk *datasort_merge(struct datasort_cfg *dcfg)
 		merged_chunk->index[total_count] = *dc;
 		merged_chunk->offset += dc->disk_size;
 		merged_chunk->count++;
+
+		// sync merged_chunk after each 1GB to control number of low priority dirty pages
+		if ((merged_chunk->offset - offset) > (1 << 30)) {
```

Comment:
Is `merged_chunk->offset` always `>=` than `offset`?

Reply:
yep, `merged_chunk->offset` is incremental and `offset` is set to `merged_chunk->offset` inside this if.